### ISSUE TITLE
Configuring stubbed __toString() does not work

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/phpunit-mock-objects#267]
There is a regression in phpunit-mock-objects 3.0 (used with PHPUnit 5) from phpunit-mock-objects 2.3 (used with PHPUnit 4):

```php
<?php
class Example
{
    public function __toString()
    {
    }
}

class Test extends PHPUnit_Framework_TestCase
{
    public function testOne()
    {
        $stub = $this->getMock(Example::class);
        $stub->method('__toString')->willReturn('string');

        $this->assertEquals('string', (string) $stub);
    }
}
```

## Expected
```
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 23 ms, Memory: 3.25Mb

OK (1 test, 1 assertion)
```

## Actual
```
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)

Time: 28 ms, Memory: 3.75Mb

There was 1 failure:

1) Test::testOne
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'string'
+''

/home/sb/Test.php:16

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

**Comment by sebastianbergmann** [comment]
Seems to be related to https://github.com/sebastianbergmann/phpunit-mock-objects/blob/master/src/Framework/MockObject/InvocationMocker.php#L105.

**Comment by sebastianbergmann** [comment]
@danez Could you have another look at this, please? Thanks!

**Comment by danez** [comment]
OK I'll do as soon as I'm home.

**Comment by stof** [comment]
#268 and #272 are both providing fixes for it

**Comment by sebastianbergmann** [comment]
Should be fixed by b28b029356e65091dfbf8c2bc4ef106ffece509e.
